File: api/src/main/java/net/blueshell/api/common/util/TimeUtil.java
package net.blueshell.api.common.util;

import java.sql.Timestamp;
import java.time.Instant;
import java.time.LocalDateTime;

public class TimeUtil {

    public static boolean hasExpired(Timestamp timestamp) {
        if (timestamp == null) return false;

        return timestamp.before(Timestamp.from(Instant.now()));
    }

    public static Timestamp of(LocalDateTime localDateTime) {
        if (localDateTime == null) return null;

        return Timestamp.valueOf(localDateTime);
    }

}

File: api/src/main/kotlin/net/blueshell/api/common/util/TimeUtil.kt
package net.blueshell.api.common.util

import java.sql.Timestamp
import java.time.Instant
import java.time.LocalDateTime

object TimeUtil {
    fun hasExpired(timestamp: Timestamp?): Boolean {
        if (timestamp == null) return false

        return timestamp.before(Timestamp.from(Instant.now()))
    }

    fun of(localDateTime: LocalDateTime?): Timestamp? {
        if (localDateTime == null) return null

        return Timestamp.valueOf(localDateTime)
    }
}
